java/src/main/java/com/lancedb/lance/Fragment.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance;
+
+/** Data Fragment. */
+public class Fragment {
+  // Only keep fragmentId for reference, so we don't need to make this
+  // object to be {@link Closable} to track Rust native object.
+  private final long fragmentId;
+
+  /** Pointer to the {@link Dataset} instance in Java. */
+  Dataset dataset;
+
+  /** Private constructor, calling from JNI. */
+  private Fragment(long fragmentId) {
+    this.fragmentId = fragmentId;
+  }
+
+  private native int countRowsNative(Dataset dataset, long fragmentId);
+
+  public long getFragmentId() {
+    return this.fragmentId;
+  }
```

Comment:
Why is fragmentId long? We only have room for 32 bits in the row address for it?

Reply:
Ok, changed to int.